sympy/functions/combinatorial/numbers.py

```diff
@@ -6,7 +6,7 @@
 Factorials, binomial coefficients and related functions are located in
 the separate 'factorials' module.
 """
-
+from math import prod as _prod
```

Comment:
I didn't know about this function. Maybe it can replace the one in `mul.py`.

Reply:
Python 3.8. Found it by accident.

Reply:
Looks like it's equivalent except that `start` is a keyword only argument:
```python
In [25]: prod([]) is math.prod([])
Out[25]: True

In [26]: prod(['asd', 2]) == math.prod(['asd', 2])
Out[26]: True

In [27]: prod([], start=-1) == math.prod([], start=-1)
Out[27]: True
```

Reply:
Why would we use `prod` but not `sum`?

Reply:
And we already have `prod` in `mul.py`:
```console
>>> help(prod)
Help on function prod in module sympy.core.mul:

prod(a, start=1)
    Return product of elements of a. Start with int 1 so if only
       ints are included then an int result is returned.
```

Reply:
@oscarbenjamin , you say "Looks like it's equivalent except that start is a keyword only argument:" but then demonstrate that you can run both mul and math `prod` with or without keyword use. I'm confused. Can you clarify?

Reply:
```
In [3]: prod([], 1)
Out[3]: 1

In [4]: import math

In [5]: math.prod([], 1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-acb528cf6bec> in <module>
----> 1 math.prod([], 1)

TypeError: prod() takes exactly 1 positional argument (2 given)
```

Reply:
doh, I missed the parens around the `math.prod(['asd', 2])` example and parsed it as `prod('asd',2)`. Thanks.